### PR TITLE
Use Gradle for build system

### PR DESCRIPTION
**Pull request by dnkoutso**
Not sure what Mr. Travis wants to work with this.

Also not sure if we want this before 2.5.1 (which might be a small release for 2.5). Doesn't bother me actually.

Tests work fine.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,6 +11,13 @@ build
...
 lib
 
+# Gradle
+.gradle
+jniLibs
+build
+local.properties
+reports
+
 # Maven
```
Comment (JakeWharton) [inline .gitignore]: Delete

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      check.dependsOn('checkstyle')
+    }
+  }
+
+  buildscript {
+    repositories {
+      mavenCentral()
+    }
+    dependencies {
+      classpath 'com.android.tools.build:gradle:1.1.0-rc1'
```
Comment (JakeWharton) [inline build.gradle]: rc2

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+ext.deps = [
+    // Android
+    supportv4: "com.android.support:support-v4:21.0.2",
+
+    // Square
+    okhttp: [
+        okhttp: "com.squareup.okhttp:okhttp:2.2.0",
+        okhttpurlconnection: "com.squareup.okhttp:okhttp-urlconnection:2.2.0",
```
Comment (JakeWharton) [inline build.gradle]: not needed

File: picasso/build.gradle [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  android {
+    lintOptions {
+      ignore 'InvalidPackage'
+    }
+  }
+}
+
+dependencies {
+  compile deps.okhttp.okhttp
```
Comment (JakeWharton) [inline picasso/build.gradle]: provided

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    }
+  }
+
+  task checkstyle(type: Checkstyle) {
+    configFile checkstyleConfig
+    source 'src/main/java'
+    ignoreFailures false
+    showViolations true
+    include '**/*.java'
+    exclude '**/gen/**'
```
Comment (JakeWharton) [inline build.gradle]: What is this for?

**Comment by JakeWharton** [comment]
I'll fix up Travis and the deploy keys later. If you haven't addressed the comments by then I'll just take care of them.

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  apply plugin: 'checkstyle'
+
+  def checkstyleConfig = rootProject.file('checkstyle.xml')
+
+  checkstyle {
+    checkstyle {
+      configFile checkstyleConfig
+      ignoreFailures true // This is for tests.
```
Comment (JakeWharton) [inline build.gradle]: Why is this set to `true` and the following thing to `false`?

**Comment by JakeWharton** [comment]
Pushed onto your branch. Still needed:

 * [ ] Generate update keys for snapshot deploy
 * [ ] Update snapshot script for Gradle
 * [ ] Include jar artifact to 'picasso' and 'picasso-pollexor' projects.

**Comment by dnkoutso** [comment]
Moving to Picasso Next.

**Comment by dnkoutso** [comment]
Just rebased to latest.

**Comment by mullender** [comment]
My pull request https://github.com/square/picasso/pull/1130 adds the gradle logic to publish the artifacts to maven.
- [ ] Generate update keys for snapshot deploy (not sure what that means)
- [x] Update snapshot script for Gradle
- [x] Include .AAR artifact to 'picasso' and 'picasso-pollexor' projects.
- [x] Include .jar artifact for sources and javadocs to 'picasso' and 'picasso-pollexor' projects.

**Comment by mullender** [comment]
Just checking to see if we should close this pull request #1130, or if there are comments that need to be addressed.

**Comment by JakeWharton** [comment]
#1409

**Pull request closed without merging**